examples/apps/cli/cli_uart.cpp

```diff
@@ -36,8 +36,8 @@
 #include <openthread/logging.h>
 
 #include "cli/cli_config.h"
-#include "common/code_utils.hpp"
 #include "common/debug.hpp"
+#include "lib/common/code_utils.hpp"
```

Comment:
How about updating `COMMON_INCLUDES` in the `src/core/CMakeLists.txt` file? We can add `${PROJECT_SOURCE_DIR}/src/lib` as one of the "include" path options.  I think with this we don't need to update core source files.

We can consider doing the same in other `CMakeLists.txt` files where we want to use the common headers.